src/main/java/core/basesyntax/Ball.java

```diff
@@ -0,0 +1,35 @@
+package core.basesyntax;
+
+import java.util.Random;
+
+public class Ball {
+    protected String color;
+    protected int number;
+
+    public Ball(String color, int number) {
+        this.number = number;
+        this.color = color;
+    }
+
+    public void setColor(String color) {
+        this.color = color;
+    }
+
+    public String getColor() {
+        return color;
+    }
+
+    public void setNumber(int number) {
+        this.number = number;
+    }
+
+    public int getNumber() {
+        return number;
+    }
+
+    @Override
+    public String toString() {
+        return "Ball { color: '" + color + "', number: " + number + " }";

```

Comment:
The 'toString' method implementation is correct, but consider overriding 'equals' and 'hashCode' as well when you override 'toString', especially for data classes. This ensures that the objects can be compared and used in collections properly.